### ISSUE TITLE
[BUG] Incorrect number of CPUs

### ISSUE THREAD

**Issue by rsj123** [autogluon/autogluon#4567]
**Bug Report Checklist**

- [x] I provided code that demonstrates a minimal reproducible example.  
- [ ] I confirmed bug exists on the latest mainline of AutoGluon via source install.  
- [x] I confirmed bug exists on the latest stable version of AutoGluon.  

**Describe the bug**
In my execution, AutoGluon incorrectly detected the actual number of available CPUs (possibly related to Docker). My container is limited to 32 vCPUs, but AutoGluon reads the number of CPUs from the host machine, resulting in an incorrect CPU allocation.
Here are some of the cpython community discussions: https://github.com/python/cpython/issues/80235#issuecomment-1093815926
**Expected behavior**
AutoGluon detected actual number of available CPUs correctly
**To Reproduce**

**Screenshots / Logs**

(autoscaler +14m43s) Error: No available node types can fulfill resource request {'CPU': 384.0}. Add suitable node types to this cluster to resolve this issue.

**Installed Versions**

<details>

INSTALLED VERSIONS
------------------
date                   : 2024-10-23
time                   : 00:41:41.426556
python                 : 3.10.8.final.0
OS                     : Linux
OS-release             : 5.4.0-153-generic
Version                : #170-Ubuntu SMP Fri Jun 16 13:43:31 UTC 2023
machine                : x86_64
processor              : x86_64
num_cores              : 384
cpu_ram_mb             : 1547877.46875
cuda version           : None
num_gpus               : 0
gpu_ram_mb             : []
avail_disk_size_mb     : 39242

accelerate             : 0.21.0
autogluon              : 1.1.1
autogluon.common       : 1.1.1
autogluon.core         : 1.1.1
autogluon.features     : 1.1.1
autogluon.multimodal   : 1.1.1
autogluon.tabular      : 1.1.1
autogluon.timeseries   : 1.1.1
boto3                  : 1.35.45
catboost               : 1.2.7
defusedxml             : 0.7.1
evaluate               : 0.4.3
fastai                 : 2.7.18
gluonts                : 0.15.1
hyperopt               : 0.2.7
imodels                : None
jinja2                 : 3.1.2
joblib                 : 1.4.2
jsonschema             : 4.21.1
lightgbm               : 4.3.0
lightning              : 2.3.3
matplotlib             : 3.6.3
mlforecast             : 0.10.0
networkx               : 3.4.2
nlpaug                 : 1.1.11
nltk                   : 3.9.1
nptyping               : 2.4.1
numpy                  : 1.24.1
nvidia-ml-py3          : 7.352.0
omegaconf              : 2.2.3
onnxruntime-gpu        : None
openmim                : 0.3.9
optimum                : 1.17.1
optimum-intel          : None
orjson                 : 3.10.9
pandas                 : 2.2.3
pdf2image              : 1.17.0
Pillow                 : 10.4.0
psutil                 : 5.9.4
pytesseract            : 0.3.10
pytorch-lightning      : 2.3.3
pytorch-metric-learning: 2.3.0
ray                    : 2.10.0
requests               : 2.28.1
scikit-image           : 0.20.0
scikit-learn           : 1.4.0
scikit-learn-intelex   : None
scipy                  : 1.12.0
seqeval                : 1.2.2
setuptools             : 65.5.0
skl2onnx               : None
statsforecast          : 1.4.0
tabpfn                 : None
tensorboard            : 2.11.2
text-unidecode         : 1.3
timm                   : 0.9.16
torch                  : 2.3.1
torchmetrics           : 1.2.1
torchvision            : 0.18.1
tqdm                   : 4.64.1
transformers           : 4.40.2
utilsforecast          : 0.0.10
vowpalwabbit           : None
xgboost                : 2.0.3

</details>

**Comment by Innixma** [comment]
We implemented a solution using joblib for AutoGluon 1.4 release that should address this issue. Please open a new issue if the problem persists in AG 1.4. Cheers!
